### DISCUSSION TITLE
Partial manual mock re-export

### DISCUSSION THREAD

**Discussion by segevfiner** [vitest-dev/vitest#3718] [Q&A]
I have some code from Jest which does something like this in a manual mock:
```ts
import { jest } from '@jest/globals';

const mod = jest.createMockFromModule('../../src/module') as {
  doStuff: object;
};

mod.doStuff = jest.fn(() => Promise.resolve([]));
export = mod;
```

And I'm not sure how to migrate this to Vitest. In particular `export =` which is a CJS only TypeScript specific syntax. And if there is a better way to handle the types in this module.

**Comment by sheremet-va** [comment]
Technically it's possible to do `module.exports = mod` in Vitest because it supports bundler module resolution

**Reply by sheremet-va** [reply]
And `jest.createMockFromModule` can be replaced with `vi.importMock`

**Reply by segevfiner** [reply]
So `export = mod;` doesn't work but `module.exports = mod;` does. `vi.importMock` doesn't work though, it just reimports the manual mock, instead of creating a mock from the actual mock as if the manual mock wasn't there like `createMockFromModule` does. Is there anything else or is this a missing feature in vitest currently and I'll have to resort to manually defining the entire module.
